cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -18,72 +18,240 @@ package loadbalancers
+type loadBalancingMode string
+
+const (
+	// Utilization determines how the traffic load is spread based on the
+	// utilization of instances.
+	loadBalancingModeUtilization = loadBalancingMode("UTILIZATION")
+
+	// Connection determines how the traffic load is spread based on the
+	// total number of connections that a backend can handle. This is
+	// only mode available for passthrough Load Balancers.
+	loadBalancingModeConnection = loadBalancingMode("CONNECTION")
+
+	loadBalanceTrafficInternal = "INTERNAL"
+)
+
+// Reconcile reconcile cluster control-plane loadbalancer components.
 func (s *Service) Reconcile(ctx context.Context) error {
 	log := log.FromContext(ctx)
 	log.Info("Reconciling loadbalancer resources")
+
+	// Creates instance groups used by load balancer(s)
 	instancegroups, err := s.createOrGetInstanceGroups(ctx)
 	if err != nil {
 		return err
 	}
 
-	healthcheck, err := s.createOrGetHealthCheck(ctx)
+	lbSpec := s.scope.LoadBalancer()
+	lbType := ptr.Deref(lbSpec.LoadBalancerType, infrav1.ExternalLoadBalancerOnly)
+	// Create a Global External Proxy Load Balancer by default
+	if lbType == infrav1.ExternalLoadBalancerOnly || lbType == infrav1.DualLoadBalancer {
+		if err = s.createExternalLoadBalancer(ctx, lbType, instancegroups); err != nil {
+			return err
+		}
+	}
+
+	// Create a Regional Internal Passthrough Load Balancer if configured
+	if lbType == infrav1.DualLoadBalancer || lbType == infrav1.InternalLoadBalancerOnly {
+		name := infrav1.InternalRoleTagValue
+		if lbSpec.InternalLoadBalancer != nil {
+			name = ptr.Deref(lbSpec.InternalLoadBalancer.Name, infrav1.InternalRoleTagValue)
+		}
+		if err = s.createInternalLoadBalancer(ctx, name, lbType, instancegroups); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Delete deletes cluster control-plane loadbalancer components.
+func (s *Service) Delete(ctx context.Context) error {
+	log := log.FromContext(ctx)
+	var externalLBErr, internalLBErr error
+	lbSpec := s.scope.LoadBalancer()
+	lbType := ptr.Deref(lbSpec.LoadBalancerType, infrav1.ExternalLoadBalancerOnly)
+	if lbType == infrav1.ExternalLoadBalancerOnly || lbType == infrav1.DualLoadBalancer {
+		externalLBErr = s.deleteExternalLoadBalancer(ctx)
+	}
+
+	if lbType == infrav1.DualLoadBalancer || lbType == infrav1.InternalLoadBalancerOnly {
+		name := infrav1.InternalRoleTagValue
+		if lbSpec.InternalLoadBalancer != nil {
+			name = ptr.Deref(lbSpec.InternalLoadBalancer.Name, infrav1.InternalRoleTagValue)
+		}
+		internalLBErr = s.deleteInternalLoadBalancer(ctx, name)
+	}
+	if externalLBErr != nil {
+		if err := s.deleteInstanceGroups(ctx); err != nil {
+			log.Error(err, "Error deleting instancegroup")
+		}
+		return externalLBErr
+	}
+	if internalLBErr != nil {
+		if err := s.deleteInstanceGroups(ctx); err != nil {
+			log.Error(err, "Error deleting instancegroup")
+		}
+		return internalLBErr
+	}
+
+	// Delete shared instancegroups last
+	return s.deleteInstanceGroups(ctx)
```

Comment:
Instead of doing this what do you think if we attempted deleting the instance groups no matter the externalLBErr/internalLBErr, and then aggregate all the errors and return them?